zebra-chain/src/transaction/serialize.rs

```diff
@@ -300,6 +300,23 @@ impl ZcashDeserialize for Transaction {
                     joinsplit_data,
                 })
             }
+            (5, false) => {
+                let id = reader.read_u32::<LittleEndian>()?;
+                if id != TX_V5_VERSION_GROUP_ID {
+                    return Err(SerializationError::Parse("expected TX_V5_VERSION_GROUP_ID"));
+                }
+                Ok(Transaction::V5 {
+                    tx_in: Vec::zcash_deserialize(&mut reader)?,
+                    tx_out: Vec::zcash_deserialize(&mut reader)?,
+                    lock_time: LockTime::zcash_deserialize(&mut reader)?,
+                    expiry_height: block::Height(reader.read_u32::<LittleEndian>()?),
+                    rest: {
+                        let mut buffer = Vec::new();
+                        reader.read_to_end(&mut buffer)?;
+                        buffer
+                    },
```

Comment:
Let's follow the pattern for V4 transactions and use separate named `let` bindings for every field:
https://github.com/ZcashFoundation/zebra/pull/1824/files/fdfc8d6276dff1242ef2f0ba1861d44562b09b7d..331805284c7bd265a0935677103e0e7b635fca3a#diff-ea762b004cd64770c3f4900c77d53620d0723dac1ab023443542de4f55e85398R265

That will help when we add a lot more fields.